### ISSUE TITLE
About preparation of Validation Set

### ISSUE THREAD

**Issue by XL19951002** [awslabs/gluon-ts#968]
For estimator.train(training_data=xxx, validation_data=xxx), should validation_data be prepared the same way as training_data? I saw that the only difference between ValidationLoader and TrainingLoader is `cyclic` field and __iter__ method. I assume that they use the exact same sampler when loading the data. So according to my understanding, I should be very careful to set the starting date in gluonts dataset in order to avoid overlapping with the training segments. For example:

                           length 20       |  length 8
               |--------------------|--------|
               |               training      |  valid    |
               Set `prediction_length=3` and `context_length=10`.

Under this settings, for training set we just use first 20 points to construct the dataset. However, for validation set, in order to strictly avoid overlapping with the training segments to prevent overfitting and use last 8 data points for validation. Should we construct validation set like this?
                      length  20        | length 8
            |--------------------|--------|
    validation set: |----------|--------|
                              length 10 | length 8
Basically, if we want to ensure that every data point during the validation period is covered, we need to leave number of data points which is equal to `context_length=10` before the validation period. And also if we leave more than `context_length=10`, let's say 11, then it is likely that ValidationLoader will generate samples which is also coming from the training set. Does this example clarifies the expected behavior of validation set?
